text/1000-migrations.rst

```diff
@@ -76,6 +80,41 @@ whether there are untracked changes in the local schema that require a
 new migration.
 
 
+Migration History
+=================
+
+The migration history is defined as a linear sequence of migrations (although
+this might change in the future to support merging migration histories).
+Each migration has a unique identifier that is a hash derived from the token
+stream of the migration script and the hash of the migration parent.  The
+hashes are computed by the ``edgedb migrate`` command starting from the
+latest committed migration (if any).  It is an error to amend an
+already-committed migration.
```

Comment:
Also need to add that when a database is created, it has a single migration with a null-hash, e.g. a migration with a sole `CREATE DATABASE db;` command.

Reply:
> migration with a sole `CREATE DATABASE db;` command.

Migrations cannot contain `DATABASE` commands (also `ROLE`, `MIGRATION`, `TRANSACTION`, etc).

Do we really need an explicit nil migration?  An empty parent would serve the same purpose.

Reply:
> Migrations cannot contain DATABASE commands (also ROLE, MIGRATION, TRANSACTION, etc).

Alright, let's not break the rules here. I guess an implicit null-migration is OK.